src/spaceObjects/spaceship.cpp

```diff
@@ -216,6 +217,17 @@ SpaceShip::~SpaceShip()
 
 void SpaceShip::applyTemplateValues()
 {
+    class_name = ship_template->getClass();
+
+    // If this ship is a station, default to the appropriate comms script and
+    // callsign, and increase its avoidance radius.
+    if (class_name == "Station" && game_server)
```

Comment:
Instead of looking at the class, I think looking at the TemplateType is the proper way to go.